src/main/java/com/epam/flightscontroltool/service/airplane/AirplaneService.java

```diff
@@ -0,0 +1,102 @@
+package com.epam.flightscontroltool.service.airplane;
+
+import com.epam.flightscontroltool.controller.dto.airplane.AirplaneDto;
+import com.epam.flightscontroltool.controller.exception.airplane.CompanyNotFoundException;
+import com.epam.flightscontroltool.controller.exception.airplane.AirplaneNotFoundException;
+import com.epam.flightscontroltool.domain.airplane.entity.Airplane;
+import com.epam.flightscontroltool.domain.airplane.mapper.AirplaneMapper;
+import com.epam.flightscontroltool.repository.airplane.CompanyRepository;
+import com.epam.flightscontroltool.repository.airplane.AirplaneRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class AirplaneService {
+
+    private final AirplaneRepository airplaneRepository;
+    private final CompanyRepository companyRepository;
+
+    public Page<AirplaneDto> getAll(Long companyId, int pageNumber, int pageSize) {
+        return (companyId == null) ? getAll(pageNumber, pageSize) : getAllByCompanyId(companyId, pageNumber, pageSize);
+    }
+
+    public Page<AirplaneDto> getAll(int pageNumber, int pageSize) {
+        Pageable pageRequest = PageRequest.of(pageNumber, pageSize, Sort.by("id"));
+        Page<Airplane> page = airplaneRepository.findAll(pageRequest);
+        return page.map(this::convertToAirplaneDto);
+    }
+
+    public Page<AirplaneDto> getAllByCompanyId(Long companyId, int pageNumber, int pageSize) {
+        Pageable pageRequest = PageRequest.of(pageNumber, pageSize, Sort.by("id"));
+        Page<Airplane> page = airplaneRepository.getAllByCompanyId(companyId, pageRequest);
+        return page.map(this::convertToAirplaneDto);
+    }
+
+    public AirplaneDto getAirplaneDetailsById(Long id) {
+        Airplane airplane = airplaneRepository.findById(id)
+                .orElseThrow(() -> new AirplaneNotFoundException(id));
+        return convertToAirplaneDto(airplane);
+    }
+
+    @Transactional
+    public AirplaneDto save(AirplaneDto airplaneDto) {
```

Comment:
"create". "Save" is a little confusing here as method name

Reply:
done, everywhere